x/stakeibc/keeper/msg_server_claim_undelegated_tokens.go

```diff
@@ -4,75 +4,75 @@ import (
 		k.Logger(ctx).Error(errMsg)
-		return nil, sdkerrors.Wrap(types.ErrInvalidHostZone, errMsg)
+		return nil, sdkerrors.Wrapf(types.ErrInvalidUserRedemptionRecord, "could not get user redemption record: %s", userRedemptionRecordKey)
+	}
+
+	// check that the record is claimable
+	if !userRedemptionRecord.GetIsClaimable() {
+		errMsg := fmt.Sprintf("User redemption record %s is not claimable", userRedemptionRecord.Id)
+		k.Logger(ctx).Error(errMsg)
+		return nil, sdkerrors.Wrapf(types.ErrInvalidUserRedemptionRecord, "user redemption record is not claimable: %s", userRedemptionRecordKey)
 	}
+
 	// grab necessary fields to construct ICA call
-	var msgs []sdk.Msg
-	redemptionAccount, found := k.GetRedemptionAccount(ctx, hostZone)
+	hostZone, found := k.GetHostZone(ctx, msg.HostZoneId)
 	if !found {
-		errMsg := fmt.Sprintf("Redemption account not found for host zone %s", msg.HostZone)
+		errMsg := fmt.Sprintf("Host zone %s not found", msg.HostZoneId)
 		k.Logger(ctx).Error(errMsg)
 		return nil, sdkerrors.Wrap(types.ErrInvalidHostZone, errMsg)
 	}
-	// go through the desired number of records and claim them
-	// TODO make this a parameter
-	MAX_CLAIMS_LIMIT := 50
-	numRecordsToClaim := min(int(msg.MaxClaims), min(len(hostZone.ClaimableRecordIds), MAX_CLAIMS_LIMIT))
-	for i := 0; i < numRecordsToClaim; i++ {
-		record, found := k.RecordsKeeper.GetUserRedemptionRecord(ctx, hostZone.ClaimableRecordIds[i])
-		if !found {
-			errMsg := fmt.Sprintf("User redemption record %s not found on host zone %s", hostZone.ClaimableRecordIds[i], hostZone.ChainId)
-			k.Logger(ctx).Error(errMsg)
-			return nil, sdkerrors.Wrap(types.ErrInvalidHostZone, errMsg)
-		}
-		errMsg := fmt.Sprintf("FromAddress %s, ToAddress %s", redemptionAccount.Address, record.Receiver)
-		k.Logger(ctx).Error(errMsg)
-		errMsg = fmt.Sprintf("Claimable record ids %s", hostZone.ClaimableRecordIds)
-		k.Logger(ctx).Error(errMsg)
-		errMsg = fmt.Sprintf("Amount %d", record.Amount)
+	redemptionAccount, found := k.GetRedemptionAccount(ctx, hostZone)
+	if !found {
+		errMsg := fmt.Sprintf("Redemption account not found for host zone %s", msg.HostZoneId)
 		k.Logger(ctx).Error(errMsg)
-
-		// TODO: how should we handle invalid records?
-
-		msgs = append(msgs, &bankTypes.MsgSend{
-			FromAddress: redemptionAccount.Address,
-			ToAddress:   record.Receiver,
-			Amount:      sdk.NewCoins(sdk.NewInt64Coin(record.Denom, int64(record.Amount))),
-		})
+		return nil, sdkerrors.Wrap(types.ErrInvalidHostZone, errMsg)
 	}
-	// TODO we should do some error handling here, in case this call fails
-	err := k.SubmitTxs(ctx, hostZone.GetConnectionId(), msgs, *redemptionAccount)
+
+	var msgs []sdk.Msg
+	msgs = append(msgs, &bankTypes.MsgSend{
+		FromAddress: redemptionAccount.Address,
+		ToAddress:   userRedemptionRecord.Receiver,
+		Amount:      sdk.NewCoins(sdk.NewInt64Coin(userRedemptionRecord.Denom, int64(userRedemptionRecord.Amount))),
```

Comment:
I've never used `NewInt64Coin`, looks like it's just a wrapper around `NewCoin`, should be fine https://github.com/cosmos/cosmos-sdk/blob/411c272d01df89cdf67c34afbba8d4188ab99320/types/coin.go#L14-L33